src/pymor/models/symplectic.py

```diff
@@ -0,0 +1,141 @@
+
+
+class QuadraticHamiltonianModel(InstationaryModel):
+    """Generic class for quadratic Hamiltonian systems.
+
+    This class describes Hamiltonian systems given by the equations::
+
+        ∂_t u(t, μ) = J * H_op(t, μ) * u(t, μ) + J * h(t, μ)
+            u(0, μ) = x_0(μ)
+
+    for t in [0,T], where H_op is a linear time-dependent |Operator|,
+    J is a canonical Poisson matrix, h is a (possibly) time-dependent
+    vector-like |Operator|, and x_0 the initial data.
+    The right-hand side of the Hamiltonian equation is J times the
+    gradient of the Hamiltonian
+
+        Ham(u, t, μ) = 1/2* u * H_op(t, μ) * u + u * h(t, μ)
+
+    Parameters
+    ----------
+    T
+        The final time T.
+    initial_data
+        The initial data `x_0`. Either a |VectorArray| of length 1 or
+        (for the |Parameter|-dependent case) a vector-like |Operator|
+        (i.e. a linear |Operator| with `source.dim == 1`) which
+        applied to `NumpyVectorArray(np.array([1]))` will yield the
+        initial data for a given |Parameter|.
+    H_op
+        The |Operator| H_op.
+    h
+        The state-independet part of the Hamiltonian h.
+    time_stepper
+        The :class:`time-stepper <pymor.algorithms.timestepping.TimeStepper>`
+        to be used by :meth:`~pymor.models.interface.Model.solve`.
+        Alternatively, the parameter nt can be specified to use the
+        :class:`implicit midpoint rule <pymor.algorithms.timestepping.ImplicitMidpointTimeStepper>`.
+    nt
+        If time_stepper is `None` and nt is specified, the
+        :class:`implicit midpoint rule <pymor.algorithms.timestepping.ImplicitMidpointTimeStepper>`
+        as time_stepper.
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each
+        intermediate vector that is calculated is returned.
+    output_functional
+        |Operator| mapping a given solution to the model output. In many applications,
+        this will be a |Functional|, i.e. an |Operator| mapping to scalars.
+        This is not required, however.
+    visualizer
+        A visualizer for the problem. This can be any object with
+        a `visualize(U, m, ...)` method. If `visualizer`
+        is not `None`, a `visualize(U, *args, **kwargs)` method is added
+        to the model which forwards its arguments to the
+        visualizer's `visualize` method.
+    name
+        Name of the model.
+    """
+
+    def __init__(self, T, initial_data, H_op, h=None, time_stepper=None, nt=None, num_values=None,
+                 output_functional=None, visualizer=None, name=None):
+        assert isinstance(H_op, Operator) and H_op.linear and H_op.range == H_op.source \
+               and H_op.range.dim % 2 == 0
+
+        # interface to use ImplicitMidpointTimeStepper via parameter nt
+        if (time_stepper is None) != (nt is None):
+            raise ValueError('Either specify time_stepper or nt (not both)')
```

Comment:
```suggestion
            raise ValueError('Either specify time_stepper or nt (not both)')
        if time_stepper is None:
            time_stepper = ImplicitMidpointTimeStepper(nt)
```

Ah sorry, my suggestion actually removed the creation of the default time stepper ...